src/components/shared/formItems/ProjectHandle.tsx

```diff
@@ -17,29 +17,32 @@ export default function ProjectHandle({
   hideLabel,
   formItemProps,
   onValueChange,
-  value,
+  initialValue,
   requireState,
   returnValue,
 }: {
-  onValueChange: (val: string) => void
-  value?: string | BigNumber
-  requireState?: 'exists' | 'notExist'
+  onValueChange?: (val: string) => void
+  initialValue?: string | BigNumber
+  requireState?: 'exists' | 'notExist' // whether the handle is required to already exist or not.
   returnValue?: 'id' | 'handle'
 } & FormItemExt) {
   const {
     theme: { colors },
   } = useContext(ThemeContext)
   const [inputContents, setInputContents] = useState<string>()
-
+  const [handleLoading, setHandleLoading] = useState<boolean>(false)
```

Comment:
Adding a loading state lets us display a spinner in the input field.

Ideally we would debounce the input so we don't re-render every input type. Perhaps this can be a follow-up issue.

Reply:
debouncing would be great here, lots of unnecessary calls right now. we can make that a separate issue

Reply:
https://github.com/jbx-protocol/juice-interface/issues/243 👍